### ISSUE TITLE
Multithread copy not setting modtime correctly on Windows

### ISSUE THREAD

**Issue by gilles007-** [rclone/rclone#3994]
**What is the problem you are having with rclone?**
download loop.

**What is your rclone version (output from rclone version)**
rclone v1.51.0
- os/arch: windows/amd64
- go version: go1.13.7

**Which OS you are using and how many bits (eg Windows 7, 64 bit)**
Windows 10, x64

**Which cloud storage system are you using? (eg Google Drive)**
Google Suite

**The command you were trying to run (eg rclone copy /tmp remote:tmp)**
"rclone.exe" sync "remote:/folder/" "f:\folder" --bwlimit 250M --config "rclone.conf" -v

**A log from the command with the -vv flag (eg output from rclone -vv copy /tmp remote:tmp)**
Multi-thread Copied (replaced existing)

hello,
i got a problem today with synching a remote (google suite) to a local drive.
i was running an old rclone version, don't remember the version.
`"rclone.exe" sync "remote:/folder/" "f:\folder" --bwlimit 250M --config "rclone.conf" -v`
my local drive was empty, it was a complete restore.
it took 24h, job was still running  and then i started to see messages "Multi-thread Copied (replaced existing)".
previously, the job was at 99%, and sudently it go back to 88%.
i checked size of all files in the cloud and my local drives, size was the same.
but some files was always downloaded again each time i start the job again, and it should not.
i tried the last version of rclone but it was the same.
then, i realize that it was all videos files.
i understand, with the "Multi-thread Copied (replaced existing)" message that it was relative to reassembling files so i add --multi-thread-streams=0
now it's ok

**Comment by gilles007-** [comment]
job is still running but i see that all downloaded files are now with the correct date

**Comment by animosity22** [comment]
Can you reproduce the issue and lists the steps with 1 file rather than syncing large amounts of files?

I tried testing with one file and can't reproduce the date being wrong with multi threaded downloads:

Download
```
felix@gemini:~$ rclone lsl GD:jellyfish-400-mbps-4k-uhd-hevc-10bit.mkv
1504953150 2019-12-22 13:57:58.621000000 jellyfish-400-mbps-4k-uhd-hevc-10bit.mkv
felix@gemini:~$ rclone copy GD:jellyfish-400-mbps-4k-uhd-hevc-10bit.mkv . -vv
2020/02/22 13:58:53 DEBUG : rclone: Version "v1.51.0" starting with parameters ["rclone" "copy" "GD:jellyfish-400-mbps-4k-uhd-hevc-10bit.mkv" "." "-vv"]
2020/02/22 13:58:53 DEBUG : Using config file from "/opt/rclone/rclone.conf"
2020/02/22 13:58:54 DEBUG : jellyfish-400-mbps-4k-uhd-hevc-10bit.mkv: Need to transfer - File not found at Destination
2020/02/22 13:58:54 DEBUG : jellyfish-400-mbps-4k-uhd-hevc-10bit.mkv: Starting multi-thread copy with 4 parts of size 358.812M
2020/02/22 13:58:54 DEBUG : jellyfish-400-mbps-4k-uhd-hevc-10bit.mkv: multi-thread copy: stream 1/4 (0-376242176) size 358.812M starting
2020/02/22 13:58:54 DEBUG : jellyfish-400-mbps-4k-uhd-hevc-10bit.mkv: multi-thread copy: stream 2/4 (376242176-752484352) size 358.812M starting
2020/02/22 13:58:54 DEBUG : jellyfish-400-mbps-4k-uhd-hevc-10bit.mkv: multi-thread copy: stream 4/4 (1128726528-1504953150) size 358.798M starting
2020/02/22 13:58:54 DEBUG : jellyfish-400-mbps-4k-uhd-hevc-10bit.mkv: multi-thread copy: stream 3/4 (752484352-1128726528) size 358.812M starting
2020/02/22 13:59:08 DEBUG : jellyfish-400-mbps-4k-uhd-hevc-10bit.mkv: multi-thread copy: stream 1/4 (0-376242176) size 358.812M finished
2020/02/22 13:59:08 DEBUG : jellyfish-400-mbps-4k-uhd-hevc-10bit.mkv: multi-thread copy: stream 3/4 (752484352-1128726528) size 358.812M finished
2020/02/22 13:59:09 DEBUG : jellyfish-400-mbps-4k-uhd-hevc-10bit.mkv: multi-thread copy: stream 4/4 (1128726528-1504953150) size 358.798M finished
2020/02/22 13:59:11 DEBUG : jellyfish-400-mbps-4k-uhd-hevc-10bit.mkv: multi-thread copy: stream 2/4 (376242176-752484352) size 358.812M finished
2020/02/22 13:59:11 DEBUG : jellyfish-400-mbps-4k-uhd-hevc-10bit.mkv: Finished multi-thread copy with 4 parts of size 358.812M
2020/02/22 13:59:54 INFO  :
Transferred:   	    1.402G / 1.402 GBytes, 100%, 24.016 MBytes/s, ETA 0s
Transferred:            0 / 1, 0%
Elapsed time:        59.7s
Transferring:
 *      jellyfish-400-mbps-4k-uhd-hevc-10bit.mkv:100% /1.402G, 5.617M/s, 0s

2020/02/22 14:00:06 DEBUG : jellyfish-400-mbps-4k-uhd-hevc-10bit.mkv: MD5 = 9d462dd04bf1f85ed21f8d9a51c709eb OK
2020/02/22 14:00:06 INFO  : jellyfish-400-mbps-4k-uhd-hevc-10bit.mkv: Multi-thread Copied (new)
2020/02/22 14:00:06 INFO  :
Transferred:   	    1.402G / 1.402 GBytes, 100%, 19.956 MBytes/s, ETA 0s
Transferred:            1 / 1, 100%
Elapsed time:      1m11.9s

2020/02/22 14:00:06 DEBUG : 12 go routines active
2020/02/22 14:00:06 DEBUG : rclone: Version "v1.51.0" finishing with parameters ["rclone" "copy" "GD:jellyfish-400-mbps-4k-uhd-hevc-10bit.mkv" "." "-vv"]
```
Date on file is good
```
felix@gemini:~$ ls -al jellyfish-400-mbps-4k-uhd-hevc-10bit.mkv
-rw-rw-r-- 1 felix felix 1504953150 Dec 22 13:57 jellyfish-400-mbps-4k-uhd-hevc-10bit.mkv
```

Trying to copy again
```
felix@gemini:~$ rclone copy GD:jellyfish-400-mbps-4k-uhd-hevc-10bit.mkv . -vv
2020/02/22 14:00:19 DEBUG : rclone: Version "v1.51.0" starting with parameters ["rclone" "copy" "GD:jellyfish-400-mbps-4k-uhd-hevc-10bit.mkv" "." "-vv"]
2020/02/22 14:00:19 DEBUG : Using config file from "/opt/rclone/rclone.conf"
2020/02/22 14:00:19 DEBUG : GD: Loaded invalid token from config file - ignoring
2020/02/22 14:00:19 DEBUG : GD: Saved new token in config file
2020/02/22 14:00:20 DEBUG : jellyfish-400-mbps-4k-uhd-hevc-10bit.mkv: Size and modification time the same (differ by 0s, within tolerance 1ms)
2020/02/22 14:00:20 DEBUG : jellyfish-400-mbps-4k-uhd-hevc-10bit.mkv: Unchanged skipping
2020/02/22 14:00:20 INFO  :
Transferred:   	         0 / 0 Bytes, -, 0 Bytes/s, ETA -
Checks:                 1 / 1, 100%
Elapsed time:         0.0s

2020/02/22 14:00:20 DEBUG : 7 go routines active
2020/02/22 14:00:20 DEBUG : rclone: Version "v1.51.0" finishing with parameters ["rclone" "copy" "GD:jellyfish-400-mbps-4k-uhd-hevc-10bit.mkv" "." "-vv"]
```

Can you make sure to run similar things and check the date/time on the file and use "-vv" when you are running the commands to share the full output.

**Comment by gilles007-** [comment]
weird, because i clearly have a date bug.
i forgot to say that i use crypt. 
i'm going to do the same test like you and post result

**Comment by animosity22** [comment]
Crypt wouldn't matter, but I ran through that as well:

```
felix@gemini:~$ rm jellyfish-400-mbps-4k-uhd-hevc-10bit.mkv
felix@gemini:~$ rclone copy gcrypt:jellyfish-400-mbps-4k-uhd-hevc-10bit.mkv . -vv
2020/02/22 14:11:06 DEBUG : rclone: Version "v1.51.0" starting with parameters ["rclone" "copy" "gcrypt:jellyfish-400-mbps-4k-uhd-hevc-10bit.mkv" "." "-vv"]
2020/02/22 14:11:06 DEBUG : Using config file from "/opt/rclone/rclone.conf"
2020/02/22 14:11:07 DEBUG : jellyfish-400-mbps-4k-uhd-hevc-10bit.mkv: Need to transfer - File not found at Destination
2020/02/22 14:11:07 DEBUG : jellyfish-400-mbps-4k-uhd-hevc-10bit.mkv: Starting multi-thread copy with 4 parts of size 358.812M
2020/02/22 14:11:07 DEBUG : jellyfish-400-mbps-4k-uhd-hevc-10bit.mkv: multi-thread copy: stream 4/4 (1128726528-1504953150) size 358.798M starting
2020/02/22 14:11:07 DEBUG : jellyfish-400-mbps-4k-uhd-hevc-10bit.mkv: multi-thread copy: stream 3/4 (752484352-1128726528) size 358.812M starting
2020/02/22 14:11:07 DEBUG : jellyfish-400-mbps-4k-uhd-hevc-10bit.mkv: multi-thread copy: stream 2/4 (376242176-752484352) size 358.812M starting
2020/02/22 14:11:07 DEBUG : jellyfish-400-mbps-4k-uhd-hevc-10bit.mkv: multi-thread copy: stream 1/4 (0-376242176) size 358.812M starting
2020/02/22 14:11:21 DEBUG : jellyfish-400-mbps-4k-uhd-hevc-10bit.mkv: multi-thread copy: stream 1/4 (0-376242176) size 358.812M finished
2020/02/22 14:11:21 DEBUG : jellyfish-400-mbps-4k-uhd-hevc-10bit.mkv: multi-thread copy: stream 4/4 (1128726528-1504953150) size 358.798M finished
2020/02/22 14:11:21 DEBUG : jellyfish-400-mbps-4k-uhd-hevc-10bit.mkv: multi-thread copy: stream 2/4 (376242176-752484352) size 358.812M finished
2020/02/22 14:11:22 DEBUG : jellyfish-400-mbps-4k-uhd-hevc-10bit.mkv: multi-thread copy: stream 3/4 (752484352-1128726528) size 358.812M finished
2020/02/22 14:11:22 DEBUG : jellyfish-400-mbps-4k-uhd-hevc-10bit.mkv: Finished multi-thread copy with 4 parts of size 358.812M
2020/02/22 14:11:22 INFO  : jellyfish-400-mbps-4k-uhd-hevc-10bit.mkv: Multi-thread Copied (new)
2020/02/22 14:11:22 INFO  :
Transferred:   	    1.402G / 1.402 GBytes, 100%, 100.894 MBytes/s, ETA 0s
Transferred:            1 / 1, 100%
Elapsed time:        14.2s

2020/02/22 14:11:22 DEBUG : 11 go routines active
2020/02/22 14:11:22 DEBUG : rclone: Version "v1.51.0" finishing with parameters ["rclone" "copy" "gcrypt:jellyfish-400-mbps-4k-uhd-hevc-10bit.mkv" "." "-vv"]
felix@gemini:~$ ls -al jellyfish-400-mbps-4k-uhd-hevc-10bit.mkv
-rw-rw-r-- 1 felix felix 1504953150 Dec 22 13:57 jellyfish-400-mbps-4k-uhd-hevc-10bit.mkv
felix@gemini:~$ rclone copy gcrypt:jellyfish-400-mbps-4k-uhd-hevc-10bit.mkv . -vv
2020/02/22 14:11:29 DEBUG : rclone: Version "v1.51.0" starting with parameters ["rclone" "copy" "gcrypt:jellyfish-400-mbps-4k-uhd-hevc-10bit.mkv" "." "-vv"]
2020/02/22 14:11:29 DEBUG : Using config file from "/opt/rclone/rclone.conf"
2020/02/22 14:11:30 DEBUG : jellyfish-400-mbps-4k-uhd-hevc-10bit.mkv: Size and modification time the same (differ by 0s, within tolerance 1ms)
2020/02/22 14:11:30 DEBUG : jellyfish-400-mbps-4k-uhd-hevc-10bit.mkv: Unchanged skipping
2020/02/22 14:11:30 INFO  :
Transferred:   	         0 / 0 Bytes, -, 0 Bytes/s, ETA -
Checks:                 1 / 1, 100%
Elapsed time:         0.0s

2020/02/22 14:11:30 DEBUG : 5 go routines active
2020/02/22 14:11:30 DEBUG : rclone: Version "v1.51.0" finishing with parameters ["rclone" "copy" "gcrypt:jellyfish-400-mbps-4k-uhd-hevc-10bit.mkv" "." "-vv"]

```

**Comment by gilles007-** [comment]
ok, so this is my 2007 file in the cloud :
```
"rclone" lsl "google_suite_crypt_id_perso:/Animes/Bip Bip et Coyote/test/" --config "rclone.conf"
733788160 2007-04-01 05:26:14.000000000 coy.avi
```
destination folder is empty :
```
F:\animes\Bip Bip et Coyote\test>dir
 Le volume dans le lecteur F s’appelle Animes
 Le numéro de série du volume est 9285-6AF3

 Répertoire de F:\animes\Bip Bip et Coyote\test

22/02/2020  20:20    <DIR>          .
22/02/2020  20:20    <DIR>          ..
               0 fichier(s)                0 octets
               2 Rép(s)  1 280 734 547 968 octets libres
```
now, this is the copy from cloud to drive :
```
"rclone.exe" sync "google_suite_crypt_id_perso:/Animes/Bip Bip et Coyote/test/" "f:\Animes\Bip Bip et Coyote\test" --bwlimit 250M --config "rclone.conf" -vv
2020/02/22 20:24:35 DEBUG : rclone: Version "v1.51.0" starting with parameters ["rclone.exe" "sync" "google_suite_crypt_id_perso:/Animes/Bip Bip et Coyote/test/" "f:\\Animes\\Bip Bip et Coyote\\test" "--bwlimit" "250M" "--config" "rclone.conf" "-vv"]
2020/02/22 20:24:35 DEBUG : Using config file from "D:\\OneDrive - CMN\\tools\\rclone\\rclone.conf"
2020/02/22 20:24:35 INFO  : Starting bandwidth limiter at 250MBytes/s
2020/02/22 20:24:41 INFO  : Local file system at //?/f:/Animes/Bip Bip et Coyote/test: Waiting for checks to finish
2020/02/22 20:24:41 INFO  : Local file system at //?/f:/Animes/Bip Bip et Coyote/test: Waiting for transfers to finish
2020/02/22 20:24:41 DEBUG : coy.avi: Starting multi-thread copy with 2 parts of size 349.938M
2020/02/22 20:24:41 DEBUG : coy.avi: multi-thread copy: stream 2/2 (366936064-733788160) size 349.857M starting
2020/02/22 20:24:41 DEBUG : coy.avi: multi-thread copy: stream 1/2 (0-366936064) size 349.938M starting
2020/02/22 20:25:08 DEBUG : coy.avi: multi-thread copy: stream 1/2 (0-366936064) size 349.938M finished
2020/02/22 20:25:11 DEBUG : coy.avi: multi-thread copy: stream 2/2 (366936064-733788160) size 349.857M finished
2020/02/22 20:25:19 DEBUG : coy.avi: Finished multi-thread copy with 2 parts of size 349.938M
2020/02/22 20:25:19 INFO  : coy.avi: Multi-thread Copied (new)
2020/02/22 20:25:19 INFO  : Waiting for deletions to finish
2020/02/22 20:25:19 INFO  :
Transferred:      699.795M / 699.795 MBytes, 100%, 18.398 MBytes/s, ETA 0s
Transferred:            1 / 1, 100%
Elapsed time:        38.0s

2020/02/22 20:25:19 DEBUG : 9 go routines active
2020/02/22 20:25:19 DEBUG : rclone: Version "v1.51.0" finishing with parameters ["rclone.exe" "sync" "google_suite_crypt_id_perso:/Animes/Bip Bip et 
Coyote/test/" "f:\\Animes\\Bip Bip et Coyote\\test" "--bwlimit" "250M" "--config" "rclone.conf" "-vv"]
```

destination folder have now 1 file but with an incorrect date, 2020 :
```
F:\animes\Bip Bip et Coyote\test>dir
 Le volume dans le lecteur F s’appelle Animes
 Le numéro de série du volume est 9285-6AF3

 Répertoire de F:\animes\Bip Bip et Coyote\test

22/02/2020  20:24    <DIR>          .
22/02/2020  20:24    <DIR>          ..
22/02/2020  20:25       733 788 160 coy.avi
               1 fichier(s)      733 788 160 octets
               2 Rép(s)  1 280 000 757 760 octets libres
```

**Comment by gilles007-** [comment]
it's an NTFS drive, no compression.
if i mount the google suite storage to a drive letter and copy a file to this drive, date is correct

**Comment by gilles007-** [comment]
i made a sync command, but i got same error date with a copy command

**Comment by animosity22** [comment]
And for completeness, if you run the same copy with the same file with multi threaded on 0, you get the proper date, correct?

**Comment by gilles007-** [comment]
yes, date is correct.
i run a job with --multi-thread-streams=0 and leaving the file on drive :
```
"rclone.exe" sync "google_suite_crypt_id_perso:/Animes/Bip Bip et Coyote/test/" "f:\Animes\Bip Bip et Coyote\test" --bwlimit 250M --multi-thread-streams=0 --config "rclone.conf" -vv
2020/02/22 20:38:38 DEBUG : rclone: Version "v1.51.0" starting with parameters ["rclone.exe" "sync" "google_suite_crypt_id_perso:/Animes/Bip Bip et Coyote/test/" "f:\\Animes\\Bip Bip et Coyote\\test" "--bwlimit" "250M" "--multi-thread-streams=0" "--config" "rclone.conf" "-vv"]
2020/02/22 20:38:38 DEBUG : Using config file from "D:\\OneDrive - CMN\\tools\\rclone\\rclone.conf"
2020/02/22 20:38:38 INFO  : Starting bandwidth limiter at 250MBytes/s
2020/02/22 20:38:44 INFO  : Local file system at //?/f:/Animes/Bip Bip et Coyote/test: Waiting for checks to finish
2020/02/22 20:38:44 DEBUG : coy.avi: Modification times differ by 113055h59m5.3592914s: 2007-04-01 03:26:14 +0000 UTC, 2020-02-22 20:25:19.3592914 +0100 CET
2020/02/22 20:38:44 INFO  : Local file system at //?/f:/Animes/Bip Bip et Coyote/test: Waiting for transfers to finish
2020/02/22 20:39:20 INFO  : coy.avi: Copied (replaced existing)
2020/02/22 20:39:20 INFO  : Waiting for deletions to finish
2020/02/22 20:39:20 INFO  :
Transferred:      699.795M / 699.795 MBytes, 100%, 19.734 MBytes/s, ETA 0s
Checks:                 1 / 1, 100%
Transferred:            1 / 1, 100%
Elapsed time:        35.4s

2020/02/22 20:39:20 DEBUG : 7 go routines active
2020/02/22 20:39:20 DEBUG : rclone: Version "v1.51.0" finishing with parameters ["rclone.exe" "sync" "google_suite_crypt_id_perso:/Animes/Bip Bip et Coyote/test/" "f:\\Animes\\Bip Bip et Coyote\\test" "--bwlimit" "250M" "--multi-thread-streams=0" "--config" "rclone.conf" "-vv"]
```
```
F:\animes\Bip Bip et Coyote\test>dir
 Le volume dans le lecteur F s’appelle Animes
 Le numéro de série du volume est 9285-6AF3

 Répertoire de F:\animes\Bip Bip et Coyote\test

22/02/2020  20:30    <DIR>          .
22/02/2020  20:30    <DIR>          ..
01/04/2007  04:26       733 788 160 coy.avi
               1 fichier(s)      733 788 160 octets
               2 Rép(s)  1 280 000 753 664 octets libres
```

**Comment by gilles007-** [comment]
you made a copy test, i made a sync test

**Comment by gilles007-** [comment]
but i got same error date with a copy

**Comment by gilles007-** [comment]
note that i'm not running with admin rights under windows.
drive has been formated from an mmc running with admin rights and i suppose default permissions

**Comment by ncw** [comment]
This looks like it is a Windows only problem - I can't reproduce it on Linux....

Multithread copy calls `SetModTime` on the local object after the file has been written in  a very similar way to non multithread copy.

However the file is still open at the time the `SetModTime` is called - I wonder if that is the difference...

Can you try this? I think it should fix the problem - the tests pass on windows so hopefully it will!

https://beta.rclone.org/branch/v1.51.0-046-g50127d53-fix-3994-multithread-time-beta/

**Comment by gilles007-** [comment]
yes, date is correct

**Comment by ncw** [comment]
I've merged this to master now which means it will be in [the latest beta](https://beta.rclone.org) in 15-30 mins and released in v1.52

I'll most likely put it in a 1.51.1 point release too
